plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestPartitionFields.java

```diff
@@ -127,14 +127,47 @@ public void testConflicts()
         assertParseName(List.of("col", "col_year", "col_year_2"), TimestampType.withZone(), List.of("year(col)", "col_year_2"), List.of("col_year_3", "col_year_2"));
     }
 
+    @Test
+    public void testConflictsWithPartitionName()
+    {
+        Schema schema = new Schema(
+                NestedField.required(1, "order_key", LongType.get()),
+                NestedField.optional(2, "comment", StringType.get()));
+
+        PartitionSpec.Builder builder = PartitionSpec.builderFor(schema)
+                .truncate("comment", 10)
+                .bucket("order_key", 10);
+        List<PartitionField> exitingPartitionFields = builder.build()
+                .fields();
+
+        // size matches the existing partition specification
+        assertParseName(List.of("order_key"), TimestampType.withZone(), exitingPartitionFields, List.of("bucket(order_key, 10)"), List.of("order_key_bucket"));
```

Comment:
it is confusing why `TimestampType.withZone()` is used here when the field `order_key` in the schema definition has the type `LongType.get()`